### ISSUE TITLE
environment file not being read by zookeeper 

### ISSUE THREAD

**Issue by sprasad09** [deric/puppet-zookeeper#16]
Hi,

We are using the this puppet module and are running into an issue where the zookeeper doesn't seem to consider the environment file and only reads the zookeeper-env.sh file. The is causing zookeeper to not start. 

We tried to symlink the zookeeper-env.sh to environment but then again it doesn't seem to read anything from it. 

Adding the values below as you would expect in a fresh install of zookeeper works.

```
export ZOO_LOG_DIR=/var/log/zookeeper
export JAVA_HOME=/usr/lib/jvm/j2sdk1.7-oracle/jre 
```

Is there a different init file that the zookeeper needs to start from so it can read from the environment file.
 Please help!

Thanks!

-shashank

**Comment by deric** [comment]
Which distribution do you use (and ZooKeeper package is from your distribution?). E.g. on Debian you should specify environment variables in `/etc/default/zookeeper` which is sourced before starting the service.

**Comment by imriz** [comment]
@deric I have the same problem with Centos 6. The issue is that /usr/libexec/zkEnv.sh has something like this inside:
{code}
if [ -f "${ZOOCFGDIR}/zookeeper-env.sh" ]; then
  . "${ZOOCFGDIR}/zookeeper-env.sh"
fi
{code}

Maybe on Debian-like systems, it tries to load /etc/zookeeper/conf/environment instead, but that is not the case with Redhat-like systems.

**Comment by deric** [comment]
@imriz chould you check if ZooKeeper on Centos 6 reads file `/etc/zookeeper/conf/java.env`?

`java.env` should be in ZooKeeper config directory with content like this:
```
export ZOO_LOG_DIR=/var/log/zookeeper
export JAVA_HOME=...
export JVMFLAGS="-Xms512m -Xmx512m"
```
I'd like to avoid managing `zkEnv.sh` by Puppet. It is very different for each version/distribution.

**Comment by deric** [comment]
Debian package of ZooKeeper 3.4.5+dfsg-2 also contains `/etc/zookeeper/conf/environment` which is sourced on service start. So, there's quite many places where we can define the same stuff :(

**Comment by cwneal** [comment]
Hi!

Having similar issues on CentOS 6, using Cloudera repo from CDH5 with this class definition:

  class { 'zookeeper':
    servers   => ['n16.example.com'],
    service_provider    => 'init',
    manage_service_file => true,
    datastore => '/disk1/zookeeper',

    # This one on ssd
    datalogstore => '/disk2/zookeeper',
    repo => 'cloudera',
    cdhver => '5',
    packages             => ['zookeeper', 'zookeeper-server'],
    service_name         => 'zookeeper-server',
    initialize_datastore => true,
    java_opts => '-Xms32m -Xmx32m -Xss256k -Djava.awt.headless=true -server -XX:+ParallelRefProcEnabled -XX:-ResizePLAB -XX:ParallelGCThreads=13 -XX:+UseCompressedOops -XX:+UseG1GC -XX:MaxGCPauseMillis=100 -XX:+DisableExplicitGC -verbose:gc -Xloggc:/var/log/zookeeper/gc.log -XX:+PrintGCDetails -XX:+PrintGCDateStamps'
  }

The service definition file "zookeeper-server" sources this environment file:

[ -e "$ZOOCFGDIR/java.env" ] && . "$ZOOCFGDIR/java.env"

But in the /etc/zookeeper/conf directory, I have these files:

root@n16[98]:/etc/zookeeper/conf> ls -l
total 16
-rw-r--r-- 1 zookeeper zookeeper 1067 Dec  2 16:51 environment
-rw-r--r-- 1 zookeeper zookeeper 2818 Dec  2 16:28 log4j.properties
-rw-r--r-- 1 zookeeper zookeeper    2 Dec  2 16:28 myid
-rw-r--r-- 1 zookeeper zookeeper 2998 Dec  2 16:28 zoo.cfg

No java.env.... :)  

I'm trying to set the java_opts parameter, which ends up in the "environment" file, but that's not referenced anywhere.

Any suggestions?

Many thanks!
Chris

**Comment by deric** [comment]
@cwneal Any time soon. I've modified the defaults, so that on RedHat `java.env` will be used. It's rather hard to find out which ZooKeeper version is running (ZooKeeper sources `java.env` since 3.4.0), so using distributions (releases) it's currently our best guess.

**Comment by deric** [comment]
Closing this relic issue, should be fixed in [0.7.0](https://github.com/deric/puppet-zookeeper/releases/tag/v0.7.0). Please open a new issue with exact version and distribution if you run into any problem.
